crates/vm/levm/src/precompiles.rs

```diff
@@ -173,11 +174,27 @@ fn identity(
 }
 
 fn sha2_256(
-    _calldata: &Bytes,
-    _gas_for_call: U256,
-    _consumed_gas: &mut U256,
+    calldata: &Bytes,
+    gas_for_call: U256,
+    consumed_gas: &mut U256,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    let data_size: u64 = calldata
+        .len()
+        .try_into()
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let cost = sha2_256_cost(data_size)?;
+    if gas_for_call < cost {
+        return Err(VMError::PrecompileError(PrecompileError::NotEnoughGas));
+    }
+
+    *consumed_gas = consumed_gas
+        .checked_add(cost)
+        .ok_or(PrecompileError::GasConsumedOverflow)?;
```

Comment:
Maybe we can abstract this behavior into a function that receives `gas_for_call`, `consumed_gas` and the `cost`, sending the `consumed_gas` as mutable.

Reply:
Agree we have to do this in every precompile so a dedicated function would be nice

Reply:
Done in [Abstract gas consumption to a function](https://github.com/lambdaclass/ethrex/pull/1520/commits/a4235ec36fc5ad0f6a86106d3f486a63a305867d)